File: practice11/11.cpp
//https://www.nowcoder.com/practice/2aa32b378a024755a3f251e75cbf233a?tpId=37&tqId=21252&tPage=2&rp=&ru=/ta/huawei&qru=/ta/huawei/question-ranking
#include<iostream>
#include<string>
using namespace std;
void result1(string &s);
void result2(string &s);
int main()
{
	string s1, s2;
	while (cin >> s1 >> s2)
	{
		result1(s1);
		result2(s2);
	}
}
void result1(string &s)
{
	int i, m;
	m = s.length();
	for (i = 0; i < m; i++)
	{
		if (s[i] == 'Z')
		{
			s[i] = 'a';
			continue;
		}
		if (s[i] == 'z')
		{
			s[i] = 'A';
			continue;
		}
		if (s[i] == '9')
		{
			s[i] = '0';
			continue;
		}
		if (s[i] >= 'A'&&s[i] <= 'Y')
		{
			s[i] = s[i] + 33;
			continue;
		}
		if (s[i] >= 'a'&&s[i] <= 'y')
		{
			s[i] = s[i] - 31;
			continue;
		}
		if (s[i] >= '0'&&s[i] <= '8')
		{
			s[i] = s[i] + 1;
			continue;
		}
	}
	cout << s << endl;
}
void result2(string &s)
{
	int i, m;
	m = s.length();
	for (i = 0; i < m; i++)
	{
		if (s[i] == 'a')
		{
			s[i] = 'Z';
			continue;
		}
		if (s[i] == 'A')
		{
			s[i] = 'z';
			continue;
		}
		if (s[i] == '0')
		{
			s[i] = '9';
			continue;
		}
		if (s[i] >= 'B'&&s[i] <= 'Z')
		{
			s[i] = s[i] + 31;
			continue;
		}
		if (s[i] >= 'b'&&s[i] <= 'z')
		{
			s[i] = s[i] - 33;
			continue;
		}
		if (s[i] >= '1'&&s[i] <= '9')
		{
			s[i] = s[i] - 1;
			continue;
		}
	}
	cout << s << endl;
}

File: practice32/11.java


//https://www.nowcoder.com/practice/839f681bf36c486fbcc5fcb977ffe432?tpId=85&tqId=29844&tPage=1&rp=1&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n=sc.nextInt();
        String[] str=new String[n];
        for(int i=0;i<n;i++){
            str[i]=sc.next();
        }
        if(Lexicographically(str)&&Lengths(str)){
            System.out.println("both");
        }
        else if(!Lexicographically(str)&&!Lengths(str)){
            System.out.println("none");
        }
        else if(Lexicographically(str)){
            System.out.println("lexicographically");
        }
        else if(Lengths(str)){
            System.out.println("lengths");
        }
    }

    static boolean Lexicographically(String[] str){
        for(int i=0;i<str.length-1;i++){
            if(str[i].compareTo(str[i+1])>0){
                return false;
            }
        }
        return true;
    }

    static boolean Lengths(String[] str){
        for(int i=0;i<str.length-1;i++){
            if(str[i].length()>str[i+1].length()){
                return false;
            }
        }
        return true;
    }
}